src/main/java/dev/blaauwendraad/masker/json/KeyContainsMasker.java

```diff
@@ -164,38 +161,103 @@ public byte[] mask(byte[] input) {
             }
             // else, if nothing has matched, continue to the next value
             // it may happen that we don't mask anything if number / boolean masking is disabled or the value is null
-
-            maskingState.backtrackCurrentJsonPath();
         }
 
         ValueMaskingUtil.flushReplacementOperations(maskingState);
 
         return maskingState.getMessage();
     }
 
+    /**
+     * Masks the array element in place.
+     *
+     * @param maskingState the masking state an index prior to the element start index
+     */
+    private void maskArrayElementInPlace(MaskingState maskingState, KeyMaskingConfig keyMaskingConfig) {
+        maskingState.incrementCurrentIndex();
```

Comment:
Wouldn't it make more sense to do this outside of calling this method and saying what you step over (one line of doc basically?)

Reply:
It is good that all "mask" methods follow the same convention of starting one index prior to the value, so it would be a bit confusing if one method does not conform to it.

Reply:
Makes sense